App crashes when launched without internet

<!--
Oh no, a bug! It happens. Thanks for reporting an issue with NewPipe. To make it easier for us to help you please enter detailed information in the template we have provided below. If a section isn't relevant, just delete it, though it would be helpful to still provide as much detail as possible.
-->

<!-- IF YOU DON'T FILL IN THE TEMPLATE PROPERLY, YOUR ISSUE IS LIABLE TO BE CLOSED. If you feel tired/lazy right now, open your issue some other time. We'll wait. -->

<!-- The comments between these brackets won't show up in the submitted issue (as you can see in the Preview). -->

### Checklist
<!-- The first box has been checked for you to show you how it is done. -->

- [x] I am using the latest version - 0.20.3 <!-- Check https://github.com/TeamNewPipe/NewPipe/releases -->
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. <!-- Seriously, check. O_O -->
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
<!--
1. Go to '...'
2. Press on '....'
3. Swipe down to '....'
-->

1. Close NewPipe (if open)
2. Remove the app from Recents
3. Disconnect the device from the internet
4. Launch NewPipe

<!-- If you can't cause the bug to show up again reliably (and hence don't have a proper set of steps to give us), please still try to give as many details as possible on how you think you encountered the bug. -->



### Actual behaviour
<!-- Tell us what happens with the steps given above. -->

App crashes.
(Unfortunately, NewPipe has stopped.)

### Expected behavior
<!-- Tell us what you expect to happen. -->

In the previous release (v0.20.2), the app would open without internet.

### Screen recording

![final-4920](https://user-images.githubusercontent.com/65857597/99670912-45898200-2a97-11eb-94dc-8aa6190c8676.gif)

### Device info

 - Android version/Custom ROM version: Android 6.0.1
 - Device model: Samsung SM-G550FY


I could reproduce this on 2 devices.

Same thing happening with me 
Report from the app:

NewPipe crashes if opened without internet connection.
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Country:__ GB
* __Content Language:__ en-GB
* __App Language:__ en_GB
* __Service:__ none
* __Version:__ 0.20.3
* __OS:__ Linux Android 5.1.1 - 22
<details><summary><b>Crash log </b></summary><p>

```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NullPointerException: Callable returned null
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:281)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:124)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5930)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1405)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1200)
Caused by: java.lang.NullPointerException: Callable returned null
	at java.util.Objects.requireNonNull(Objects.java:109)
	at io.reactivex.internal.functions.ObjectHelper.requireNonNull(ObjectHelper.java:39)
	at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:43)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)

```
</details>
<hr>

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Country:__ KR
* __Content Language:__ ko
* __App Language:__ en
* __Service:__ none
* __Version:__ 0.20.3
* __OS:__ Linux Android 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NullPointerException: Callable returned null
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:281)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:124)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:8167)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
Caused by: java.lang.NullPointerException: Callable returned null
	at java.util.Objects.requireNonNull(Objects.java:228)
	at io.reactivex.internal.functions.ObjectHelper.requireNonNull(ObjectHelper.java:39)
	at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:43)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```
</details>
<hr>

For what it's worth, I was able to reproduce this as well. I opened the app while on WiFi, turned WiFi off (as it seemed too slow and I wanted to switch to mobile data) while the "trending" screen was still loading, then the app crashed. Unfortunately I seem to be unable to reliably reproduce this issue this way.

## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Country:__ DE
* __Content Language:__ en
* __App Language:__ en_US
* __Service:__ none
* __Version:__ 0.20.3
* __OS:__ Linux Android 10 - 29
<details><summary><b>Crash log </b></summary><p>

```
io.reactivex.exceptions.OnErrorNotImplementedException: The exception was not handled due to missing onError handler in the subscribe() method call. Further reading: https://github.com/ReactiveX/RxJava/wiki/Error-Handling | java.lang.NullPointerException: Callable returned null
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:77)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:281)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:255)
	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:124)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7697)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
Caused by: java.lang.NullPointerException: Callable returned null
	at java.util.Objects.requireNonNull(Objects.java:228)
	at io.reactivex.internal.functions.ObjectHelper.requireNonNull(ObjectHelper.java:39)
	at io.reactivex.internal.operators.observable.ObservableFromCallable.subscribeActual(ObservableFromCallable.java:43)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```
</details>
<hr>

--- a/app/src/main/java/org/schabi/newpipe/App.java
+++ b/app/src/main/java/org/schabi/newpipe/App.java
@@ -8,6 +8,7 @@ import android.os.Build;
 import android.util.Log;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.multidex.MultiDexApplication;
 import androidx.preference.PreferenceManager;
 
@@ -67,7 +68,7 @@ public class App extends MultiDexApplication {
     protected static final String TAG = App.class.toString();
     private static App app;
 
-    private Disposable disposable = null;
+    @Nullable private Disposable disposable = null;
 
     @NonNull
     public static App getApp() {

--- a/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
+++ b/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
@@ -198,24 +198,36 @@ public final class CheckForNewAppVersion {
                 })
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(response -> {
-                    // Parse the json from the response.
-                    try {
-                        final JsonObject githubStableObject = JsonParser.object().from(response)
-                                .getObject("flavors").getObject("github").getObject("stable");
+                .subscribe(
+                        response -> {
+                            // Parse the json from the response.
+                            try {
+                                final JsonObject githubStableObject = JsonParser.object()
+                                        .from(response).getObject("flavors").getObject("github")
+                                        .getObject("stable");
 
-                        final String versionName = githubStableObject.getString("version");
-                        final int versionCode = githubStableObject.getInt("version_code");
-                        final String apkLocationUrl = githubStableObject.getString("apk");
+                                final String versionName = githubStableObject
+                                        .getString("version");
+                                final int versionCode = githubStableObject
+                                        .getInt("version_code");
+                                final String apkLocationUrl = githubStableObject
+                                        .getString("apk");
 
-                        compareAppVersionAndShowNotification(app, versionName, apkLocationUrl,
-                                versionCode);
-                    } catch (final JsonParserException e) {
-                        // connectivity problems, do not alarm user and fail silently
-                        if (DEBUG) {
-                            Log.w(TAG, Log.getStackTraceString(e));
-                        }
-                    }
-                });
+                                compareAppVersionAndShowNotification(app, versionName,
+                                        apkLocationUrl, versionCode);
+                            } catch (final JsonParserException e) {
+                                // connectivity problems, do not alarm user and fail silently
+                                if (DEBUG) {
+                                    Log.w(TAG, Log.getStackTraceString(e));
+                                }
+                            }
+                        },
+                        throwable -> {
+                            // connectivity problems, do not alarm user and fail silently
+                            if (DEBUG) {
+                                Log.i(TAG, "Could not get NewPipe API: network problem");
+                                Log.i(TAG, Log.getStackTraceString(throwable));
+                            }
+                        });
     }
 }

--- a/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
+++ b/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
@@ -22,13 +22,11 @@ import com.grack.nanojson.JsonObject;
 import com.grack.nanojson.JsonParser;
 import com.grack.nanojson.JsonParserException;
 
-import org.schabi.newpipe.extractor.exceptions.ReCaptchaException;
 import org.schabi.newpipe.report.ErrorActivity;
 import org.schabi.newpipe.report.ErrorInfo;
 import org.schabi.newpipe.report.UserAction;
 
 import java.io.ByteArrayInputStream;
-import java.io.IOException;
 import java.io.InputStream;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
@@ -186,15 +184,7 @@ public final class CheckForNewAppVersion {
                     }
 
                     // Make a network request to get latest NewPipe data.
-                    try {
-                        return DownloaderImpl.getInstance().get(NEWPIPE_API_URL).responseBody();
-                    } catch (IOException | ReCaptchaException e) {
-                        // connectivity problems, do not alarm user and fail silently
-                        if (DEBUG) {
-                            Log.w(TAG, Log.getStackTraceString(e));
-                        }
-                        return null;
-                    }
+                    return DownloaderImpl.getInstance().get(NEWPIPE_API_URL).responseBody();
                 })
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
@@ -218,15 +208,14 @@ public final class CheckForNewAppVersion {
                             } catch (final JsonParserException e) {
                                 // connectivity problems, do not alarm user and fail silently
                                 if (DEBUG) {
-                                    Log.w(TAG, Log.getStackTraceString(e));
+                                    Log.w(TAG, "Could not get NewPipe API: invalid json", e);
                                 }
                             }
                         },
-                        throwable -> {
+                        e -> {
                             // connectivity problems, do not alarm user and fail silently
                             if (DEBUG) {
-                                Log.i(TAG, "Could not get NewPipe API: network problem");
-                                Log.i(TAG, Log.getStackTraceString(throwable));
+                                Log.w(TAG, "Could not get NewPipe API: network problem", e);
                             }
                         });
     }

--- a/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
+++ b/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
@@ -12,6 +12,7 @@ import android.net.Uri;
 import android.util.Log;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.core.app.NotificationCompat;
 import androidx.core.app.NotificationManagerCompat;
 import androidx.core.content.ContextCompat;
@@ -36,10 +37,9 @@ import java.security.cert.CertificateException;
 import java.security.cert.CertificateFactory;
 import java.security.cert.X509Certificate;
 
-import io.reactivex.Observable;
+import io.reactivex.Maybe;
 import io.reactivex.android.schedulers.AndroidSchedulers;
 import io.reactivex.disposables.Disposable;
-import io.reactivex.disposables.Disposables;
 import io.reactivex.schedulers.Schedulers;
 
 public final class CheckForNewAppVersion {
@@ -174,54 +174,51 @@ public final class CheckForNewAppVersion {
         return getCertificateSHA1Fingerprint(app).equals(GITHUB_APK_SHA1);
     }
 
-    @NonNull
+    @Nullable
     public static Disposable checkNewVersion(@NonNull final App app) {
         final SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(app);
 
         // Check if user has enabled/disabled update checking
         // and if the current apk is a github one or not.
-        if (!prefs.getBoolean(app.getString(R.string.update_app_key), true)
-                || !isGithubApk(app)) {
-            return Disposables.empty();
+        if (!prefs.getBoolean(app.getString(R.string.update_app_key), true) || !isGithubApk(app)) {
+            return null;
         }
 
-        return Observable.fromCallable(() -> {
-            if (!isConnected(app)) {
-                return null;
-            }
+        return Maybe
+                .fromCallable(() -> {
+                    if (!isConnected(app)) {
+                        return null;
+                    }
 
-            // Make a network request to get latest NewPipe data.
-            try {
-                return DownloaderImpl.getInstance().get(NEWPIPE_API_URL).responseBody();
-            } catch (IOException | ReCaptchaException e) {
-                // connectivity problems, do not alarm user and fail silently
-                if (DEBUG) {
-                    Log.w(TAG, Log.getStackTraceString(e));
-                }
-            }
-
-            return null;
-        })
+                    // Make a network request to get latest NewPipe data.
+                    try {
+                        return DownloaderImpl.getInstance().get(NEWPIPE_API_URL).responseBody();
+                    } catch (IOException | ReCaptchaException e) {
+                        // connectivity problems, do not alarm user and fail silently
+                        if (DEBUG) {
+                            Log.w(TAG, Log.getStackTraceString(e));
+                        }
+                        return null;
+                    }
+                })
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribe(response -> {
                     // Parse the json from the response.
-                    if (response != null) {
-                        try {
-                            final JsonObject githubStableObject = JsonParser.object().from(response)
-                                    .getObject("flavors").getObject("github").getObject("stable");
+                    try {
+                        final JsonObject githubStableObject = JsonParser.object().from(response)
+                                .getObject("flavors").getObject("github").getObject("stable");
 
-                            final String versionName = githubStableObject.getString("version");
-                            final int versionCode = githubStableObject.getInt("version_code");
-                            final String apkLocationUrl = githubStableObject.getString("apk");
+                        final String versionName = githubStableObject.getString("version");
+                        final int versionCode = githubStableObject.getInt("version_code");
+                        final String apkLocationUrl = githubStableObject.getString("apk");
 
-                            compareAppVersionAndShowNotification(app, versionName, apkLocationUrl,
-                                    versionCode);
-                        } catch (final JsonParserException e) {
-                            // connectivity problems, do not alarm user and fail silently
-                            if (DEBUG) {
-                                Log.w(TAG, Log.getStackTraceString(e));
-                            }
+                        compareAppVersionAndShowNotification(app, versionName, apkLocationUrl,
+                                versionCode);
+                    } catch (final JsonParserException e) {
+                        // connectivity problems, do not alarm user and fail silently
+                        if (DEBUG) {
+                            Log.w(TAG, Log.getStackTraceString(e));
                         }
                     }
                 });

--- a/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
+++ b/app/src/main/java/org/schabi/newpipe/CheckForNewAppVersion.java
@@ -58,48 +58,43 @@ public final class CheckForNewAppVersion {
      * @param application The application
      * @return String with the apk's SHA1 fingeprint in hexadecimal
      */
+    @NonNull
     private static String getCertificateSHA1Fingerprint(@NonNull final Application application) {
-        final PackageManager pm = application.getPackageManager();
-        final String packageName = application.getPackageName();
-        final int flags = PackageManager.GET_SIGNATURES;
-        PackageInfo packageInfo = null;
-
+        final PackageInfo packageInfo;
         try {
-            packageInfo = pm.getPackageInfo(packageName, flags);
+            packageInfo = application.getPackageManager().getPackageInfo(
+                    application.getPackageName(), PackageManager.GET_SIGNATURES);
         } catch (final PackageManager.NameNotFoundException e) {
             ErrorActivity.reportError(application, e, null, null,
                     ErrorInfo.make(UserAction.SOMETHING_ELSE, "none",
                             "Could not find package info", R.string.app_ui_crash));
+            return "";
         }
 
-        final Signature[] signatures = packageInfo.signatures;
-        final byte[] cert = signatures[0].toByteArray();
-        final InputStream input = new ByteArrayInputStream(cert);
-
-        X509Certificate c = null;
-
+        final X509Certificate c;
         try {
+            final Signature[] signatures = packageInfo.signatures;
+            final byte[] cert = signatures[0].toByteArray();
+            final InputStream input = new ByteArrayInputStream(cert);
             final CertificateFactory cf = CertificateFactory.getInstance("X509");
             c = (X509Certificate) cf.generateCertificate(input);
         } catch (final CertificateException e) {
             ErrorActivity.reportError(application, e, null, null,
                     ErrorInfo.make(UserAction.SOMETHING_ELSE, "none",
                             "Certificate error", R.string.app_ui_crash));
+            return "";
         }
-
-        String hexString = null;
 
         try {
             final MessageDigest md = MessageDigest.getInstance("SHA1");
             final byte[] publicKey = md.digest(c.getEncoded());
-            hexString = byte2HexFormatted(publicKey);
+            return byte2HexFormatted(publicKey);
         } catch (NoSuchAlgorithmException | CertificateEncodingException e) {
             ErrorActivity.reportError(application, e, null, null,
                     ErrorInfo.make(UserAction.SOMETHING_ELSE, "none",
                             "Could not retrieve SHA1 key", R.string.app_ui_crash));
+            return "";
         }
-
-        return hexString;
     }
 
     private static String byte2HexFormatted(final byte[] arr) {
@@ -164,10 +159,10 @@ public final class CheckForNewAppVersion {
     }
 
     private static boolean isConnected(@NonNull final App app) {
-        final ConnectivityManager cm = ContextCompat.getSystemService(app,
-                ConnectivityManager.class);
-        return cm.getActiveNetworkInfo() != null
-                && cm.getActiveNetworkInfo().isConnected();
+        final ConnectivityManager connectivityManager =
+                ContextCompat.getSystemService(app, ConnectivityManager.class);
+        return connectivityManager != null && connectivityManager.getActiveNetworkInfo() != null
+                && connectivityManager.getActiveNetworkInfo().isConnected();
     }
 
     public static boolean isGithubApk(@NonNull final App app) {

